src/CodeExampleLanguageProcessing.ts

```diff
@@ -0,0 +1,371 @@
+    canonicalValue: CanonicalLanguageValues.SHELL,
+    extension: LanguageFileExtensions.SHELL,
+  },
+  {
+    directiveValue: "shell",
+    canonicalValue: CanonicalLanguageValues.SHELL,
+    extension: LanguageFileExtensions.SHELL,
+  },
+  {
+    directiveValue: "swift",
+    canonicalValue: CanonicalLanguageValues.SWIFT,
+    extension: LanguageFileExtensions.SWIFT,
+  },
+  {
+    directiveValue: "text",
+    canonicalValue: CanonicalLanguageValues.TEXT,
+    extension: LanguageFileExtensions.TEXT,
+  },
+  {
+    directiveValue: "typescript",
+    canonicalValue: CanonicalLanguageValues.TYPESCRIPT,
+    extension: LanguageFileExtensions.TYPESCRIPT,
+  },
+  {
+    directiveValue: "undefined",
+    canonicalValue: CanonicalLanguageValues.UNDEFINED,
+    extension: LanguageFileExtensions.UNDEFINED,
+  },
+  {
+    directiveValue: "xml",
+    canonicalValue: CanonicalLanguageValues.XML,
+    extension: LanguageFileExtensions.XML,
+  },
+  {
+    directiveValue: "yaml",
+    canonicalValue: CanonicalLanguageValues.YAML,
+    extension: LanguageFileExtensions.YAML,
+  },
+  {
+    directiveValue: "json\\n :copyable: false",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: ".json",
+  },
+  {
+    directiveValue: "json\\n:copyable: true",
+    canonicalValue: CanonicalLanguageValues.JSON,
+    extension: ".json",
+  },
+];
+
+export const getLanguageDetailsForCodeNode = (
+  node: CodeNode
+): LanguageMapper => {
+  if (node.lang != null) {
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue === node.lang
+    );
+    if (mapping != undefined) {
+      return mapping;
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  } else {
+    return {
+      directiveValue: "none",
+      canonicalValue: CanonicalLanguageValues.UNDEFINED,
+      extension: LanguageFileExtensions.UNDEFINED,
+    };
+  }
+};
+
+export const getLanguageDetailsForLiteralInclude = (
+  node: DirectiveNode
+): LanguageMapper => {
+  const maybeChildNode = node.children[0] as CodeNode;
+  const maybeNodeChildLang = maybeChildNode.lang;
+  if (maybeNodeChildLang != null) {
+    const language: string = maybeNodeChildLang as string;
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue === language
+    );
+    if (mapping != undefined) {
+      return mapping;
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  } else {
+    const maybeNodeLang = node.options?.language;
+    if (maybeNodeLang != undefined) {
+      const language = maybeNodeLang;
+      const mapping = languageValueMappings.find(
+        (mapper) => mapper.directiveValue === language
+      );
+      if (mapping != undefined) {
```

Comment:
Inconsistently using `null` vs. `undefined`. Personally I would just triple-check `===` on undefined, as nothing in this codebase _should_ be using null. But if you're going to soft-check, stick with null.